external/dispatch/CMakeLists.txt

```diff
@@ -4,59 +4,62 @@
 # For license and copyright information please follow this link:
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
-add_library(external_dispatch INTERFACE IMPORTED GLOBAL)
-add_library(desktop-app::external_dispatch ALIAS external_dispatch)
+set(dispatch_loc ${third_party_loc}/dispatch)
+set(dispatch_prefix ${CMAKE_CURRENT_BINARY_DIR}/dispatch-prefix)
+set(dispatch_binary_dir ${dispatch_prefix}/src/dispatch-build)
 
 if (DESKTOP_APP_USE_PACKAGED)
     find_library(DISPATCH_LIBRARIES dispatch)
     find_path(DISPATCH_INCLUDE_DIRS dispatch/dispatch.h)
+endif()
 
-    if (DISPATCH_LIBRARIES AND DISPATCH_INCLUDE_DIRS)
-        target_include_directories(external_dispatch SYSTEM INTERFACE ${DISPATCH_INCLUDE_DIRS})
-        target_link_libraries(external_dispatch INTERFACE ${DISPATCH_LIBRARIES})
+if (NOT DISPATCH_LIBRARIES OR NOT DISPATCH_INCLUDE_DIRS)
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} ${dispatch_loc}
+            -B ${dispatch_binary_dir}
+            -GNinja
```

Comment:
CMake may be run for producing Makefiles. Is it necessary to force Ninja here?

Reply:
You can't set generator for nested run and I believe Ninja is better than make